superset-frontend/src/SqlLab/components/ScheduleQueryButton.tsx

```diff
@@ -0,0 +1,214 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { FunctionComponent, useState } from 'react';
+import Form, { FormProps, FormValidation } from 'react-jsonschema-form';
+import { Col, FormControl, FormGroup, Row } from 'react-bootstrap';
+import { t, styled } from '@superset-ui/core';
+import * as chrono from 'chrono-node';
+import ModalTrigger from 'src/components/ModalTrigger';
+import FormLabel from 'src/components/FormLabel';
+import './ScheduleQueryButton.less';
+import Button from 'src/components/Button';
+
+const validators = {
+  greater: (a: number, b: number) => a > b,
+  greater_equal: (a: number, b: number) => a >= b,
+  less: (a: number, b: number) => a < b,
+  less_equal: (a: number, b: number) => a <= b,
+};
+
+const getJSONSchema = () => {
+  const jsonSchema = window.featureFlags.SCHEDULED_QUERIES?.JSONSCHEMA;
+  // parse date-time into usable value (eg, 'today' => `new Date()`)
```

Comment:
For this feature, though, @zhaoyongjie implemented an awesome [date parser](https://github.com/apache/superset/blob/7b74eb2730a9b0d592de2a4999072f83df193b11/superset/utils/date_parser.py#L0-L1) (in Python) for time filters, do you think it would make sense to consolidate these to use the same API?